popup-wrapper.js

```diff
@@ -1,10 +1,23 @@
 chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
-    chrome.tabs.sendMessage(tabs[0].id, {
-        type: "popup"
-    },
-    (response) => {
-        if (response) {
-            //console.log(response);
+    if (tabs[0] === undefined || !tabs[0].url.startsWith('https://www.youtube.com/')) {
+        const errorMessageSpan = document.querySelector('#extErrorMessage')
+        errorMessageSpan.textContent = 'You need to be on YouTube!'
+        return
+    }
+
+    chrome.tabs.executeScript(tabs[0].id, {
+        // showPopup acutally toggles popup here
+        code: 'window.__PREVYOU_LOADED ? (showPopup(), true) : false'
+    }, res => {
+        if (!res[0]) {
```

Comment:
`executeScript` could run a script on multiple tabs, which is why it always returns an array (one result per tab) - here we know we always have one tab so we can just use `res[0]`